src/Domain/Creators/TimeDependentOptions/ExpansionMap.cpp

```diff
@@ -4,87 +4,206 @@
+  }
 }
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic pop
+#endif  // defined(__GNUC__) && !defined(__clang__)
+
+template <bool AllowSettleFoTs>
+ExpansionMapOptions<AllowSettleFoTs>::ExpansionMapOptions(
+    const std::array<double, 3>& initial_values_in,
+    double decay_timescale_outer_boundary_in,
+    double asymptotic_velocity_outer_boundary_in,
+    const Options::Context& /*context*/)
+    : decay_timescale_outer_boundary(decay_timescale_outer_boundary_in),
+      asymptotic_velocity_outer_boundary(
+          asymptotic_velocity_outer_boundary_in) {
+  initial_values = std::array{DataVector{initial_values_in[0]},
+                              DataVector{initial_values_in[1]},
+                              DataVector{initial_values_in[2]}};
+  initial_values_outer_boundary =
+      std::array{DataVector{1.0}, DataVector{0.0}, DataVector{0.0}};
+}
+
+template <bool AllowSettleFoTs>
+std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime> get_expansion(
+    const std::variant<ExpansionMapOptions<AllowSettleFoTs>, FromVolumeFile>&
+        expansion_map_options,
+    const double initial_time, const double expiration_time) {
+  const std::string name = "Expansion";
+  std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime> result{};
+
+  if (std::holds_alternative<FromVolumeFile>(expansion_map_options)) {
+    const auto& from_vol_file = std::get<FromVolumeFile>(expansion_map_options);
+    const auto volume_fot =
+        from_vol_file.retrieve_function_of_time({name}, initial_time);
+
+    // It must be either a PiecewisePolynomial or a SettleToConstant
+    const auto* pp_volume_fot =
+        dynamic_cast<domain::FunctionsOfTime::PiecewisePolynomial<2>*>(
+            volume_fot.at(name).get());
+    const auto* settle_volume_fot =
+        dynamic_cast<domain::FunctionsOfTime::SettleToConstant*>(
+            volume_fot.at(name).get());
+
+    if (UNLIKELY(pp_volume_fot == nullptr and settle_volume_fot == nullptr)) {
+      ERROR_NO_TRACE(
+          "Expansion function of time read from volume data is not a "
+          "PiecewisePolynomial<2> or a SettleToConstant. Cannot use it to "
+          "initialize the expansion map.");
+    }
+
+    if (pp_volume_fot != nullptr) {
+      result =
+          volume_fot.at(name)->create_at_time(initial_time, expiration_time);
+    } else {
+      ASSERT(AllowSettleFoTs,
+             "Expansion function of time in the volume file is a "
+             "SettleToConstant, but SettleToConstant functions of time aren't "
+             "allowed.");
+      result = volume_fot.at(name)->get_clone();
+    }
+  } else if (std::holds_alternative<ExpansionMapOptions<AllowSettleFoTs>>(
+                 expansion_map_options)) {
+    const auto& hard_coded_options =
+        std::get<ExpansionMapOptions<AllowSettleFoTs>>(expansion_map_options);
+
+    if (hard_coded_options.asymptotic_velocity_outer_boundary.has_value()) {
```

Comment:
confused that `asymptotic_velocity_outer_boundary` is not actually used below. Add a comment?